src/goto-instrument/contracts/contracts.cpp

```diff
@@ -727,6 +727,29 @@ code_contractst::create_ensures_instruction(
   return std::make_pair(std::move(ensures_program), std::move(history));
 }
 
+static const code_with_contract_typet &
+get_contract(const irep_idt &function, const namespacet &ns)
+{
+  const std::string function_str = id2string(function);
+  const symbolt &function_symbol = ns.lookup(function);
+
+  const symbolt *contract_sym;
+  if(ns.lookup("contract::" + function_str, contract_sym))
+  {
+    // no contract provided in the source or just an empty assigns clause
+    return to_code_with_contract_type(function_symbol.type);
+  }
+
+  const auto &type = to_code_with_contract_type(contract_sym->type);
+  if(type != function_symbol.type)
```

Comment:
How can this equality work if the function symbol had its contract clauses removed but the contract symbol still has them ? 

Would the check still work if in addition we removed the function body from the contract and renamed contract parameters ? 

Maybe we could just check that the number of arguments and their types are equal.

Reply:
This perhaps warrants an extra comment: the comparison works, because all the extra information is stored in comment-elements of the ireps: both the parameter identifiers as well as all the contract clauses have `ID_C_...` keys in the `irept`. Therefore, equality (but not full equality!) works out. While the body isn't there as indicated in the earlier comment, it wouldn't affect this comparison either way for it's just a comparison of the types, not the symbols' values.